### PR TITLE
Initial setup for Phase 2: Added return tokens to lexer file

### PR DESCRIPTION
**Pull request by GodfreyL3**

### CODE REVIEW

**Review by CoolGuyRy — APPROVED** [review]

File: src/lexer.lex [hunk 1]
```diff
@@ -62,63 +62,61 @@ COMMENT		(\$\$([^(\$\$)]*(\n?))*\$\$)|(\$[^(\$|\n)]*\$)
...
-{OR}			{ fprintf(yyout, "OR\n"); lineCharacterCount += yyleng;}
-{SEMICOLON}		{ fprintf(yyout, "SEMICOLON\n"); lineCharacterCount += yyleng; }
-{COMMA}			{ fprintf(yyout, "COMMA\n"); lineCharacterCount += yyleng;}
-
-{OUTPUT}		{ fprintf(yyout, "OUTPUT\n"); lineCharacterCount += yyleng;}
-{INPUT}			{ fprintf(yyout, "INPUT\n"); lineCharacterCount += yyleng;}
-
-{IDENTIFIER}   		{ fprintf(yyout, "IDENTIFIER: %s\n", yytext); lineCharacterCount += yyleng; }
-{BAD_IDENTIFIER}	{ printf("Error: Invalid identifier %s on line %d, col %d\n", yytext, lineCount + 1, lineCharacterCount + 1); lineCharacterCount += yyleng;}
+{INTEGER}       	{ return INTEGER }
```
Comment (CoolGuyRy) [inline src/lexer.lex]: fprintf(yyout, "INTEGER \n"); lineCharacterCount += yyleng; should be here. If not our lineCharacterCount will be off whenever ints are declared.

File: src/lexer.lex [hunk 2]
```diff
@@ -62,63 +62,61 @@ COMMENT		(\$\$([^(\$\$)]*(\n?))*\$\$)|(\$[^(\$|\n)]*\$)
...
+{DOUBLE}    		{ return DOUBLE }
```
Comment (CoolGuyRy) [inline src/lexer.lex]: same thing here

File: src/lexer.lex [hunk 3]
```diff
@@ -62,63 +62,61 @@ COMMENT		(\$\$([^(\$\$)]*(\n?))*\$\$)|(\$[^(\$|\n)]*\$)
...
+{NUMBER}      		{ fprintf(yyout, "NUMBER: %s\n", yytext); lineCharacterCount += yyleng; return NUMBER; }
+{DECIMAL}    		{ fprintf(yyout, "DECIMAL: %s\n", yytext); lineCharacterCount += yyleng; return DECIMAL; }
 
-
-{NEWLINE}       	{ lineCount++; lineCharacterCount = 0; }
-{WHITESPACE}    	{ lineCharacterCount += yyleng; }
-.       		{ printf("Error: Unrecognized symbol \"%s\" on line %d, col %d\n", yytext, lineCount + 1, lineCharacterCount + 1); lineCharacterCount += yyleng; }
+{NEWLINE}       	{ lineCount++; lineCharacterCount = 0; return NEWLINE; }
+{WHITESPACE}    	{ lineCharacterCount += yyleng; return WHITESPACE; }
+.       		{ printf("Error: Unrecognized symbol \"%s\" on line %d, col %d\n", yytext, lineCount + 1, lineCharacterCount + 1); lineCharacterCount += yyleng; return .;}
```
Comment (CoolGuyRy) [inline src/lexer.lex]: Is this the right way to return errors?

File: src/lexer.lex [hunk 4]
```diff
@@ -62,63 +62,61 @@ COMMENT		(\$\$([^(\$\$)]*(\n?))*\$\$)|(\$[^(\$|\n)]*\$)
...
+{AND}			{ fprintf(yyout, "AND\n"); lineCharacterCount += yyleng; return AND; }
+{OR}			{ fprintf(yyout, "OR\n"); lineCharacterCount += yyleng; return OR; }
+{SEMICOLON}		{ fprintf(yyout, "SEMICOLON\n"); lineCharacterCount += yyleng; return SEMICOLON; }
+{COMMA}			{ fprintf(yyout, "COMMA\n"); lineCharacterCount += yyleng; return COMMA; }
+
+{OUTPUT}		{ fprintf(yyout, "OUTPUT\n"); lineCharacterCount += yyleng; return OUTPUT; }
+{INPUT}			{ fprintf(yyout, "INPUT\n"); lineCharacterCount += yyleng; return INPUT; }
+
+{IDENTIFIER}   		{ fprintf(yyout, "IDENTIFIER: %s\n", yytext); lineCharacterCount += yyleng; return IDENTIFIER; }
+{BAD_IDENTIFIER}	{ printf("Error: Invalid identifier %s on line %d, col %d\n", yytext, lineCount + 1, lineCharacterCount + 1); lineCharacterCount += yyleng; return BAD_IDENTIFIER; }
```
Comment (CoolGuyRy) [inline src/lexer.lex]: Is this the right way to return errors?

**Review by CoolGuyRy — CHANGES_REQUESTED** [review]
A couple things I commented need to be addressed. But other than that lgtm!

**Review by Howie315 — APPROVED** [review]

**Pull request merged**